base/pkg/entry.jl

```diff
@@ -197,7 +197,11 @@ function clone(url::AbstractString, pkg::AbstractString)
         end
     catch err
         isdir(pkg) && Base.rm(pkg, recursive=true)
-        rethrow(err)
+        if err.msg == "Unexpected HTTP status code: 404"
```

Comment:
> I think you should also check that isa(err, GitError), since not all exceptions have a msg field.

Are you suggesting updating line 200 to something like
`if isa(err, Base.LibGit2.Error.GitError) && err.msg == "Unexpected HTTP status code: 404"`? If so, do all `GitError` exceptions have a message field?

>I wonder whether there shouldn't even be an error code which could be checked, rather than comparing the whole string. That would be more robust (but it would require more significant changes...).

How do you feel about
`if sprint(show, err) == "GitError(Code:ERROR, Class:Net, Unexpected HTTP status code: 404)"`?


> will you get a 404 if you try installing a package with the correct url when your connection is disabled, or will the error be different?

When I go offline, I get a GitError with a different message:
`ERROR: GitError(Code:ERROR, Class:Net, curl error: Couldn't resolve host 'github.com'
)`